### PR TITLE
Add guide on versions and releases

### PR DESCRIPTION
**Pull request by ekoleda-codaio**
[Staged here](https://coda-packs-sdk--1834.com.readthedocs.build/en/1834/guides/development/versions/)

Also changed "web IDE" to "web editor" throughout a few other pages.

### CODE REVIEW

File: docs/guides/development/versions.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+```ts
+export const pack = coda.newPack();
+pack.setVersion("1.0.5");
+```
+
+There are some restrictions to the version numbers you can select however:
+
+- The version numbers must increase with each new build / upload.
+- The version numbers must follow the [SemVer standard][semver].
```
Comment (jonathan-codaio) [inline docs/guides/development/versions.md:36]: Might be worth adding a simple example so people don't have to grok the spec. Can be as simple as "That is, versions must be of the form `1`, `1.2`, or `1.2.3`."

File: docs/guides/development/versions.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+## Releases
+
+Releases are used to determine which version of your Pack should be made available to your users. For example, as you are developing and testing a new feature you may create multiple new versions of your Pack, but your users will not see any of those changes until you make a new release. This allows you experiment and iterate on your code and only share that work when it's ready.
+
+A new release is created when:
+
+- You click the **Release** button in the Pack Studio web editor.
+- You click the **Release version** option in the **History** tab of the Pack Studio.
+- You run the `npx code release` command using the Pack CLI.
```
Comment (jonathan-codaio) [inline docs/guides/development/versions.md:62]: code-->coda

**Review by jonathan-codaio — APPROVED** [review]
😌 So nice to have these concepts in writing for the first time since we conceived of them like a year ago.

Probably worth having one more section that explains what actually happens in practice when you create a new release. Namely that Coda gradually sweeps through docs that currently use your pack and upgrades them to the released version. This process begins within a few minutes of creating your release, and can take anywhere between a few minutes to a few hours to work through all customer docs.

File: docs/guides/development/versions.md (hunk 1 above)
Comment (ekoleda-codaio) [reply docs/guides/development/versions.md]: Done.

File: docs/guides/development/versions.md (hunk 2 above)
Comment (ekoleda-codaio) [reply docs/guides/development/versions.md]: Done.

**Comment by ekoleda-codaio** [comment]
> Probably worth having one more section that explains what actually happens in practice when you create a new release. Namely that Coda gradually sweeps through docs that currently use your pack and upgrades them to the released version. This process begins within a few minutes of creating your release, and can take anywhere between a few minutes to a few hours to work through all customer docs.

TIL! Added a section called "Rollout".

**Pull request merged**